kakao/src/main/kotlin/com/agoda/kakao/configurator/ConfiguratorBuilder.kt

```diff
@@ -0,0 +1,98 @@
+        ((Web.WebInteraction<*>) -> WebInteractionDelegate)
+
+    init {
+        onViewInteraction { }
+        onDataInteraction { }
+        onWebInteraction { }
+    }
+
+    /**
+     * Operator that allows usage of DSL style
+     *
+     * @param function Tail lambda with receiver which is your view
+     */
+    operator fun invoke(function: ConfiguratorBuilder.() -> Unit) {
+        function(this)
+    }
+
+    fun onViewInteraction(
+        override: Boolean = false,
+        onViewInteractionDelegateBuilder: ViewInteractionDelegateBuilder.(ViewInteraction) -> Unit
+    ) {
+        viewInteractionDelegateFactory = { viewInteraction ->
+            val viewInteractionDelegateBuilder = ViewInteractionDelegateBuilder(
+                viewInteractionForBuilder = viewInteraction,
+                parentViewInteractionDelegate =
+                    if (override) parentConfig?.viewInteractionDelegateFactory?.invoke(viewInteraction)
+                    else null
+            )
+            onViewInteractionDelegateBuilder.invoke(viewInteractionDelegateBuilder, viewInteraction)
+            viewInteractionDelegateBuilder.viewInteractionDelegate
+        }
+    }
+
+    fun onDataInteraction(
+        override: Boolean = false,
+        onDataInteractionDelegateBuilder: DataInteractionDelegateBuilder.(DataInteraction) -> Unit
+    ) {
+        dataInteractionDelegateFactory = { dataInteraction ->
+            val dataInteractionDelegateBuilder = DataInteractionDelegateBuilder(
+                dataInteractionForBuilder = dataInteraction,
+                parentDataInteractionDelegate =
+                    if (override) parentConfig?.dataInteractionDelegateFactory?.invoke(dataInteraction)
+                    else null
+            )
+            onDataInteractionDelegateBuilder.invoke(dataInteractionDelegateBuilder, dataInteraction)
+            dataInteractionDelegateBuilder.dataInteractionDelegate
+        }
+    }
+
+    fun onWebInteraction(
+        override: Boolean = false,
+        onWebInteractionDelegateBuilder: WebInteractionDelegateBuilder.(Web.WebInteraction<*>) -> Unit
+    ) {
+        webInteractionDelegateFactory = { webInteraction ->
+            val webInteractionDelegateBuilder = WebInteractionDelegateBuilder(
+                webInteractionForBuilder = webInteraction,
+                parentWebInteractionDelegate =
+                    if (override) parentConfig?.webInteractionDelegateFactory?.invoke(webInteraction)
+                    else null
+            )
+            onWebInteractionDelegateBuilder.invoke(webInteractionDelegateBuilder, webInteraction)
+            webInteractionDelegateBuilder.webInteractionDelegate
+        }
+    }
+
+    fun getConfigurator(): ConfigModel {
+        return ConfigModel(
+            parentConfig,
+            viewInteractionDelegateFactory,
+            dataInteractionDelegateFactory,
+            webInteractionDelegateFactory
+        )
+    }
+
+}
```

Comment:
Missing new line

Reply:
> And also @Unlimity please don't create comments for every file about `new line`. We can figure it out after lint check adding. I suppose that I can do it on this weekend.

Yep, I am waiting tuned code style like Detekt for example